### ISSUE TITLE
NSInternalInconsistencyException - UITableView was deallocated while key value observers were still registered with it.

### ISSUE THREAD

**Issue by rankinit** [Yalantis/Persei#17]
I have  a TableViewController that uses Persei.  I push the TableViewController when I select a button on my starting view.  When I hit the back button in the navigation bar I get the following error:

Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'An instance  of class UITableView was deallocated while key value observers were still registered with it. Current observation info: <NSKeyValueObservationInfo 0x7fd003571060> (
<NSKeyValueObservance 0x7fd003576350: Observer: 0x7fd00370e800, Key path: contentOffset, Options: <New: YES, Old: NO, Prior: NO> Context: 0x103111190, Property: 0x7fd00378e3a0>

It looks like in StickyHeaderView is setting the observer here:

```
public weak var scrollView: UIScrollView! {
    willSet {
        self.scrollView?.removeObserver(self, forKeyPath: "contentOffset", context: &ContentOffsetContext)
        self.scrollView?.panGestureRecognizer.removeTarget(self, action: "handlePan:")

        appliedInsets = UIEdgeInsetsZero
    }

    didSet {
        scrollView?.addObserver(self, forKeyPath: "contentOffset", options: .Initial | .New, context: &ContentOffsetContext)
        scrollView?.panGestureRecognizer.addTarget(self, action: "handlePan:")
    }
}
```

How can I fix this issue?  Thanks!

**Comment by dimazen** [comment]
Hell, sorry for the delay, I'll take a look at the next week.

**Comment by dimazen** [comment]
Can you upload an example or some code?

**Comment by dimazen** [comment]
My rough idea is that `willMoveToSuperview` is not getting called before UITableView dealloc. A little bit unexpected. Still would like to take a look at some rough sample code. Thanks in advance.

**Comment by dimazen** [comment]
@rankinit fixed in [2.0.1](https://github.com/Yalantis/Persei/releases/tag/2.0.1)
